File: memories/server/routes/posts.ts
import express, { Request, Response } from 'express';
import { getPosts, getPost, createPost, updatePost, likePost, deletePost } from '../controllers/posts'; // Assuming the posts controllers are also written in TypeScript

const router = express.Router();

// Define routes with correct types for request and response
router.get('/', (req: Request, res: Response) => getPosts(req, res));
router.post('/', (req: Request, res: Response) => createPost(req, res));
router.get('/:id', (req: Request, res: Response) => getPost(req, res));
router.patch('/:id', (req: Request, res: Response) => updatePost(req, res));
router.delete('/:id', (req: Request, res: Response) => deletePost(req, res));
router.patch('/:id/likePost', (req: Request, res: Response) => likePost(req, res));

export default router;

File: memories/server/routes/posts.go
package routes

import (
	"github.com/gin-gonic/gin"
	"go.mongodb.org/mongo-driver/mongo"
	"your_project/controllers"
)

func PostRoutes(router *gin.Engine, postCollection *mongo.Collection) {
	// Pass the collection to the controller functions
	router.GET("/posts", func(c *gin.Context) {
		controllers.GetPosts(c, postCollection)
	})

	router.POST("/posts", func(c *gin.Context) {
		controllers.CreatePost(c, postCollection)
	})

	router.GET("/posts/:id", func(c *gin.Context) {
		controllers.GetPost(c, postCollection)
	})

	router.PATCH("/posts/:id", func(c *gin.Context) {
		controllers.UpdatePost(c, postCollection)
	})

	router.DELETE("/posts/:id", func(c *gin.Context) {
		controllers.DeletePost(c, postCollection)
	})

	router.PATCH("/posts/:id/like", func(c *gin.Context) {
		controllers.LikePost(c, postCollection)
	})
}
